framework/src/problems/FEProblemBase.C

```diff
@@ -7878,6 +7878,10 @@ FEProblemBase::addOutput(const std::string & object_type,
     if (_app.getParam<bool>("show_input") && parameters.get<bool>("output_screen"))
       parameters.set<ExecFlagEnum>("execute_input_on") = EXEC_INITIAL;
   }
+  // Need this because Checkpoint::validParams changes the default value of
+  // execute_on
+  else if (object_type == "Checkpoint")
+    exclude.push_back("execute_on");
```

Comment:
This shouldn't be needed anymore as we don't apply private parameters

Reply:
Removing this triggers the ParamError, so it seems like that call to applyParameters in FEProblemBase is still overwriting the desired value of `execute_on`.

Reply:
It was decided not to make the param private because we wanted the option to specify INITIAL and/or TIMESTEP_END.